restlicsharpclient/restliclientintegrationtest/RestClientIntegrationTests.cs

```diff
@@ -98,14 +100,45 @@ public void GetGreeting_Async()
         }
 
         [TestMethod]
+        public void GetGreeting_Async_Fail()
+        {
+            RestClient client = new RestClient(urlPrefix);
+
+            GetRequestBuilder<int, Greeting> requestBuilder = new GetRequestBuilder<int, Greeting>("/basicCollection");
+            requestBuilder.SetID(-1);
+            GetRequest<int, Greeting> request = requestBuilder.Build();
+
+            AutoResetEvent blocker = new AutoResetEvent(false);
+
+            Greeting greeting = null;
+            RestliException error = null;
+
+            RestliCallback<EntityResponse<Greeting>>.SuccessHandler successHandler = delegate (EntityResponse<Greeting> response)
+            {
+                greeting = response.element;
+                blocker.Set();
+            };
+            RestliCallback<EntityResponse<Greeting>>.ErrorHandler errorHandler = delegate (ClientErrorResponse response)
+            {
+                error = response.error;
+                blocker.Set();
+            };
+            RestliCallback<EntityResponse<Greeting>> callback = new RestliCallback<EntityResponse<Greeting>>(successHandler, errorHandler);
+
+            client.RestRequestAsync(request, callback);
+
+            blocker.WaitOne();
+
+            Assert.IsNull(greeting);
+            Assert.IsNotNull(error);
+            Assert.IsNotNull(error.details);
```

Comment:
Please also verify status code and message in error.details are value returned from server.